core/src/main/clojure/xtdb/indexer.clj

```diff
@@ -614,11 +624,25 @@
               (finally
                 (.unlock wm-lock wm-lock-stamp)))))
 
-        (while (.isBlockFull live-chunk)
-          (finish-block! this))
+        (let [force-flush-req (some? @force-flush-tx-id)
+              force-flush-match (and force-flush-req (= @force-flush-tx-id last-tx-id))]
+          (cond
+            force-flush-match
+            (log/info "received flush instruction: no newer transactions found, will flush chunk.")
+
+            force-flush-req
+            (log/info "received flush instruction: newer transaction found, so not flushing chunk."))
+
+          (when (or (.isBlockFull live-chunk)
+                    force-flush-match)
+            (loop []
+              (finish-block! this)
+              (when (.isBlockFull live-chunk)
```

Comment:
this `loop` is to keep flushing blocks if we happen to get a large transaction in that's bigger than our block size - if memory serves, `finish-block!` might not leave the live-chunk empty. so I'd assume you'll want to keep flushing blocks til it's completely empty here?

(could definitely have benefited from a comment here, apologies)

Reply:
Yes I think we need a way to tell non-empty. I would look at something like blockRowLimit and rowCount methods (removing isBlockFull), but I would welcome your guidance as these are not familiar interfaces to me.

Reply:
yeah - think this is an example of unnecessarily hiding the data that you mentioned the other day. I'd be happy with that change :+1: `blockRowLimit` then needn't live with the live-chunk, think that might already be configuration on the indexer...?

Reply:
Did this, but I put the rowsPerBlock on live-chunk, (indexer didn't have it, and to me it seems the live-chunk should be transparent about it) let me know if that is a problem.